src/cli.ts

```diff
@@ -204,17 +205,23 @@ function writeToHtml(html?: string): void {
     return;
   }
 
-  const tempName = temp.path({ suffix: '.html' });
-
-  fs.writeFileSync(tempName, html);
-
-  open(tempName, { wait: false }).catch(error => {
-    logError('Unable to open web browser.', error);
-    logError(
-      'Either run with --html, --json, --tsv, --file, or view HTML for the visualization at:'
-    );
-    logError(tempName);
-  });
+  try {
+    const info = temp.openSync({ prefix: 'sme-result-', suffix: '.html' });
+    fs.writeFileSync(info.fd, html);
+
+    open(info.path).then(child => {
+      if (child.stderr) {
+        // Catch error output from child process
+        child.stderr.once('data', (data: Buffer) => {
+          logError(
+            getErrorMessage({ code: 'CannotOpenTempFile', tempFile: info.path, error: data })
+          );
+        });
+      }
+    });
+  } catch (error) {
```

Comment:
Does `open` itself throw? A `try` / `catch` wrapped around callback code makes me nervous since exceptions in the callbacks won't get caught. Suggest either converting the `open` call to use `async` / `await` or moving the `catch` block higher up, to before `open`.

Reply:
Only if `target` is not `string`. Otherwise, it simply [returns child process pointer](https://github.com/sindresorhus/open/blob/master/index.js). I will convert `open` call to use `async` / `await`

Reply:
I don't think there could be any exceptions inside the callback though

Reply:
Done